corim/signedcorim.go

```diff
@@ -127,13 +131,16 @@ func (o *SignedCorim) FromCOSE(buf []byte) error {
 }
 
 // Sign returns the serialized signed-corim, signed by the supplied cose Signer.
-// The target SignedCorim must have its UnsignedCorim field correctly
-// populated.
-func (o *SignedCorim) Sign(signer cose.Signer) ([]byte, error) {
+// The target SignedCorim must have its UnsignedCorim field correctly populated.
+func (o *SignedCorim) Sign(signer cose.Signer, leafCert, intermediateCert []byte) ([]byte, error) {
 	if signer == nil {
 		return nil, errors.New("nil signer")
 	}
 
+	if leafCert == nil || intermediateCert == nil {
+		return nil, errors.New("nil certs")
+	}
```

Comment:
We want to continue supporting the case of signed CoRIMs that rely on out-of-band means for communicating the verification key material.

Therefore, we shouldn't return an error here; continue as usual, just avoid adding the header.